### PR TITLE
test: Improve test suite defaults

### PR DESCRIPTION
**Pull request by dcramer**
- update pytest deps (+xdist)
- bind default settings (test)
- fix pytest-xdist config

### CODE REVIEW

**Comment by dcramer** [comment]
kafka tests hang for me but guess i probably need something running i dont have

**Comment by fpacifici** [comment]
The kafka test is an integration test. It can take up to one minute the first time.

File: requirements.txt [hunk 1]
```diff
@@ -50,7 +50,7 @@ pbr==4.0.4
 petname==2.2
 pexpect==4.6.0
 pickleshare==0.7.4
-pluggy==0.7.1
+pluggy==0.13.0
```
Comment (fpacifici) [inline requirements.txt]: you can even remove this and let pytest pick it up as a dependency. None uses it directly.

Comment (dcramer) [reply requirements.txt]: I'll follow up with a proposed change to poetry which has a lockfile included, and just keep this in here for now.

File: requirements.txt [hunk 2]
```diff
@@ -59,9 +59,11 @@ pyflakes==1.6.0
 Pygments==2.2.0
 pylint==1.9.2
 pyparsing==2.2.0
-pytest==3.9.3
+pytest==5.2.4
 pytest-cov==2.5.1
+pytest-forked==1.1.3
```
Comment (tkaemming) [inline requirements.txt]: These test have a bunch of shared external state — running forked tests will likely cause a lot of test failures.

Comment (dcramer) [reply requirements.txt]: fwiw i pulled that out of pip-freeze - probably a dep of pytest-xdist

Comment (dcramer) [reply requirements.txt]: also note in other PR: i wasnt aware of pytest-watch (that it existed or what it did), and pytest-xdist dupes looponfail. could remove one of these

**Pull request merged**